app/src/main/java/com/example/sharingang/items/Item.kt

```diff
@@ -20,9 +21,10 @@ data class Item(
 
     val price: Double = 0.0,
 
-    val createdAt: Date = Date(),
+    @ServerTimestamp
+    val createdAt: Date? = null,
 
     /** ID generated by the database */
     @DocumentId
-    var id: String? = null
+    val id: String? = null
```

Comment:
"val" here would mean that id can never be reassigned no?

Reply:
I believe so but why would you need to change the id anyways? Plus that way you can use ids to compare items (as seen below) since you know they won't change

Reply:
Yes, at the beginning we used var because we had to set the id after the item was added to the database.
But I discovered a better way, we can create a copy of the item and only change the id like that: `item.copy(id = ...)`